### DISCUSSION TITLE
Quantization

### DISCUSSION THREAD

**Discussion by sunjaxx** [abey79/vpype#579] [Ideas]
Hello,

Is there any way (or appetite) to make vpype **avoid** the quantization (polylines approximation) of curves, at least for some "recipes"?

In my use case, where I'd use vpype to mainly optimize the order (and direction) of paths, I see the quantization as a limitation, since my (many?) cutters can handle (for example) bezier curves natively. 

Couldn't for example a curve line (for the purpose of optimization) simply be temporarily replaced with a **single** straight/dummy path (having the same start point, end point and direction as the curve), and once the optimization is done, **replaced back** with the original curve (eventually with its direction reversed, depending on the optimization results)? Wouldn't this also result in quicker processing times and smaller outputs?

Thanks!

**Comment by abey79** [comment]
Hey @sunjaxx. There is currently no way to avoid quantisation with vpype. Arm twisting the metadata system to carry the original SVG curve information and regurgitate it at export would *in theory* be possible, but it would require a massive amount of hacks, especially because metadata is only at layer level (not at path level). Still, some highly customised plug-in could theoretically achieve this for a very specific setup. I can't see myself investing time into this though.

Instead, my plan is to completely overhaul the data model for vpype2 and include all of the SVG curved stuff (bezier, arc, etc.). Quantisation would be postponed to whichever command actually requires it, which wouldn't include `linesort` indeed. I have no timeframe for when the vpype2 project will actually start though—let alone when it'll be completed. 😅

**Reply by sunjaxx** [reply]
Thanks, I see. Until then: are you maybe aware of any other SVG path optimization tool that does not involve quantization? :-)

**Reply by abey79** [reply]
Unfortunately, not really (I've been using vpype for that purpose 😀). I know the Axidraw CLI can do it (and export back to SVG, so no need to own one), but it might quantise as well, I'm not sure.
